src/data/mv_polynomial/basic.lean

```diff
@@ -422,7 +428,7 @@ by rw [X_pow_eq_monomial, support_monomial, if_neg (one_ne_zero' R)]
 
 @[simp] lemma support_zero : (0 : mv_polynomial σ R).support = ∅ := rfl
 
-lemma support_smul [distrib_mul_action R S₁] {a : R} {f : mv_polynomial σ S₁} :
+lemma support_smul [smul_zero_class R S₁] {a : R} {f : mv_polynomial σ S₁} :
```

Comment:
I'm pretty sure this assumes a lot about `R` that it doesn't need to; you might need to write
```suggestion
lemma support_smul {R : Type*} [smul_zero_class R S₁] {a : R} {f : mv_polynomial σ S₁} :
```
or if you like
```suggestion
lemma support_smul {S₁ : Type*} [smul_zero_class S₁ R] {a : S₁} {f : mv_polynomial σ R} :
```

Reply:
Oh it does assume a lot about `R`. I choose the second one for its context uses `mv_polynomial σ R`.